wear/src/main/java/com/android/gpstest/wear/MainActivity.kt

```diff
@@ -0,0 +1,419 @@
+
+        if (!userDeniedPermission) {
+            requestPermissionAndStartGps(this)
+        } else {
+            LibUIUtils.showLocationPermissionDialog(this)
+        }
+
+        setContent {
+            WearApp(LocationLabelAndData.locationLabelAndDataSample, signalInfoViewModel)
+        }
+    }
+
+    private fun requestPermissionAndStartGps(activity: Activity) {
+        if (PermissionUtils.hasGrantedPermissions(activity, arrayOf(ACCESS_FINE_LOCATION, ACCESS_COARSE_LOCATION))) {
+            gpsStart()
+        } else {
+            // Request permissions from the user
+            ActivityCompat.requestPermissions(
+                activity,
+                arrayOf(ACCESS_FINE_LOCATION, ACCESS_COARSE_LOCATION),
+                PermissionUtils.LOCATION_PERMISSION_REQUEST
+            )
+        }
+    }
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int, permissions: Array<String>, grantResults: IntArray
+    ) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        if (requestCode == PermissionUtils.LOCATION_PERMISSION_REQUEST) {
+            if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                userDeniedPermission = false
+                gpsStart()
+            } else {
+                userDeniedPermission = true
+            }
+        }
+    }
+
+    @ExperimentalCoroutinesApi
+    @SuppressLint("MissingPermission")
+    @Synchronized
+    private fun gpsStart() {
+        PreferenceUtils.saveTrackingStarted(true, prefs)
+
+        // Observe flows
+        observeLocationFlow()
+    }
+
+    @ExperimentalCoroutinesApi
+    private fun observeLocationFlow() {
+        // This should be a Flow and not LiveData to ensure that the Flow is active before the Service is bound
+        if (locationFlow?.isActive == true) {
+            // If we're already observing updates, don't register again
+            return
+        }
+        // Observe locations via Flow as they are generated by the repository
+        locationFlow = repository.getLocations()
+            .flowWithLifecycle(lifecycle, Lifecycle.State.STARTED)
+            .onEach {
+                //lastLocation = it
+                Log.d(TAG, "Activity location: $it")
+            }
+            .launchIn(lifecycleScope)
+    }
+
+    @Synchronized
+    private fun gpsStop() {
+        PreferenceUtils.saveTrackingStarted(false, prefs)
+        locationFlow?.cancel()
+    }
+
+    companion object {
+        private const val TAG = "MainActivity"
+    }
+}
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@Composable
+fun WearApp(satStatues: List<String>, signalInfoViewModel: SignalInfoViewModel) {
+    val gnssStatuses: List<SatelliteStatus> by signalInfoViewModel.filteredGnssStatuses.observeAsState(emptyList())
+    val location: Location by signalInfoViewModel.location.observeAsState(Location("default"))
```

Comment:
`location` is currently unused here - you'll need to update the Composables with data from this object to show `Location` data on the screen.

Reply:
> `location` is currently unused here - you'll need to update the Composables with data from this object to show `Location` data on the screen.

Thanks, I'll.